File: src/types/mail.ts
export interface MailSummary {
  id: string
  from_addr: string
  from_name: string | null
  subject: string
  received_at: string
  size_bytes: number
  has_html: boolean
  is_read: boolean
  attachment_count: number
}

export interface Mail extends MailSummary {
  message_id: string | null
  to_addrs: string[]
  text_body: string | null
  html_body: string | null
  raw_mime: string
  attachments: MailAttachment[]
}

export interface MailAttachment {
  id: string
  mail_id: string
  filename: string
  content_type: string
  size_bytes: number
}

export interface SmtpSessionLog {
  id: string
  mail_id: string | null
  remote_addr: string
  started_at: string
  ended_at: string
  transcript: string
  error: string | null
}

export type MailSearchField = 'all' | 'from' | 'to' | 'subject' | 'body' | 'attachments'

export interface MailSearchFilters {
  query: string
  field: MailSearchField
  unreadOnly: boolean
  hasAttachments: boolean
}

export interface MailSearchResult {
  items: MailSummary[]
  total_count: number
  unread_count: number
}

export interface AppConfig {
  smtp_port: number
  theme: 'light' | 'dark' | 'system'
  max_mails: number
  check_updates_on_startup: boolean
  auto_install_updates: boolean
  enable_notifications: boolean
}

File: src-tauri/src/models/mail.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MailSummary {
    pub id: String,
    pub from_addr: String,
    pub from_name: Option<String>,
    pub subject: String,
    pub received_at: String,
    pub size_bytes: u32,
    pub has_html: bool,
    pub is_read: bool,
    pub attachment_count: u32,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Mail {
    pub id: String,
    pub message_id: Option<String>,
    pub from_addr: String,
    pub from_name: Option<String>,
    pub subject: String,
    pub received_at: String,
    pub size_bytes: u32,
    pub has_html: bool,
    pub is_read: bool,
    pub attachment_count: u32,
    pub to_addrs: Vec<String>,
    pub text_body: Option<String>,
    pub html_body: Option<String>,
    pub raw_mime: String,
    pub attachments: Vec<MailAttachment>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MailAttachment {
    pub id: String,
    pub mail_id: String,
    pub filename: String,
    pub content_type: String,
    pub size_bytes: u32,
}

#[derive(Debug, Clone)]
pub struct MailAttachmentData {
    pub id: String,
    pub filename: String,
    pub content_type: String,
    pub content: Vec<u8>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SmtpSessionLog {
    pub id: String,
    pub mail_id: Option<String>,
    pub remote_addr: String,
    pub started_at: String,
    pub ended_at: String,
    pub transcript: String,
    pub error: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct MailSearchResult {
    pub items: Vec<MailSummary>,
    pub total_count: u32,
    pub unread_count: u32,
}
